prog.tex

```diff
@@ -51,16 +51,18 @@ \section{Variable definitions}
 
 \end{itemize}
 
+Note
+that the environment of a lambda expression used with define
+includes the variable so
+it can be called recursively.
+
```

Comment:
I am not a native English speaker but that not doesn't sounds good to me. What about something like:

```
Note that \hyper{expression} is evaluated in an environment that includes the \hyper{variable} being defined.
That feature allows recursive calls in lambda expressions.
```

but I feel it not perfect...

Reply:
Thanks, that is a useful suggestion.

Reply:
How about this:
```
Note that the if \hyper{expression} is a lambda expression, it is
evaluated in an environment includes the \hyper{variable} so it can be
called recursively.
```

Reply:
I like it but let suggest an other end:

```
... that includes the defined \hyper[variable} so it can be used for recursing.
```

Reply:
I think this works for me:
```
Note that if \hyper{expression} is a lambda expression, it is
evaluated in an environment that includes the defined \hyper[variable} 
so it can be used for recursion.
```

Reply:
yes indeed, it works for me too!